kalite/distributed/static/js/distributed/user/views.js

```diff
@@ -314,4 +314,64 @@ window.UserView = BaseView.extend({
     logout: function() {
         this.model.logout();
     }
-});
\ No newline at end of file
+});
+
+/* This view toggles which navbar items are displayed to each type of user */ 
+window.ToggleNavbarView = BaseView.extend ({
+
+    template: HB.template("user/navigation"),
+
+    initialize: function() {
+
+        _.bindAll(this);        
+        this.listenTo(this.model, "change", this.render);
+        $("topnav").append(this.template());
+
+    },
+
+    render: function() {
+        
+        this.facility_id = {"facility_id": this.model.get("facility_id")};
+        this.$el.html(this.template(this.facility_id));
+        
+        this.userView = new UserView({ model: this.model, el: "#topnav" });
+        $("#topnav").prepend(new AutoCompleteView().$el);
+        
+        // GUEST
```

Comment:
I feel like all of this could be collapsed into the handlebars template.

Instead of toggling 'display-none' classes, you can just use {{#if is_logged_in}} {{/if}} around navbar items, and just not include them if not needed.